tests/integration/vlan_test.py

```diff
@@ -62,6 +65,53 @@ def test_set_vlan_iface_down(eth1_up):
         assert not current_state[INTERFACES]
 
 
+def test_add_down_remove_vlan(eth1_up):
```

Comment:
How about `test_set_vlan_state_down` ?

Reply:
The original problem was that it was not possible to remove the vlan after trying to down it, so the whole cycle was important. Since there is no checkpoint rollback anymore, this is not tested and also it does not fail reliable anymore when testing it locally.

Reply:
I lost you here, sorry. I though we are checking that the `down` is working.
You want to check that `absent` is working after `down`?

Reply:
Maybe it is also both... It has been too long ago since I developed the test but at that time there was also an issue with not being able to remove a vlan after down failed. I did not yet manage to replicate this yesterday (at least not reliable). Nevertheless, the test is for the full cycle of adding, downing and removing the vlan, therefore the name suits IMHO.